noxfile.py

```diff
@@ -684,220 +681,13 @@ def notebook(session: nox.Session):
-    processes = []
-    for benchmark in benchmark_script_list:
-        process = Process(
-            target=session.run,
-            args=("python", benchmark),
-            kwargs={"env": {LOGGING_NAME_ENV_VAR: benchmark.as_posix()}},
-        )
-        process.start()
-        processes.append(process)
-
-    for process in processes:
-        process.join()
-
-    # when the environment variable is set as it is above,
-    # notebooks output a .bytesprocessed and .slotmillis report
-    # collect those reports and print a summary
-    _print_performance_report(base_path)
-
-
-def _print_performance_report(path: str):
-    """Add an informational report about http queries, bytes
-    processed, and slot time to the testlog output for purposes
-    of measuring bigquery-related performance changes.
-
-    Looks specifically for output files in subfolders of the
-    passed path. (*/*.bytesprocessed and */*.slotmillis)
-    """
-    print("---BIGQUERY USAGE REPORT---")
-    results_dict = {}
-    bytes_reports = sorted(Path(path).rglob("*.bytesprocessed"))
-    for bytes_report in bytes_reports:
-        with open(bytes_report, "r") as bytes_file:
-            filename = bytes_report.relative_to(path).with_suffix("")
-            lines = bytes_file.read().splitlines()
-            query_count = len(lines)
-            total_bytes = sum([int(line) for line in lines])
-            results_dict[filename] = [query_count, total_bytes]
-        os.remove(bytes_report)
-
-    millis_reports = sorted(Path(path).rglob("*.slotmillis"))
-    for millis_report in millis_reports:
-        with open(millis_report, "r") as millis_file:
-            filename = millis_report.relative_to(path).with_suffix("")
-            lines = millis_file.read().splitlines()
-            total_slot_millis = sum([int(line) for line in lines])
-            results_dict[filename] += [total_slot_millis]
-        os.remove(millis_report)
-
-    cumulative_queries = 0
-    cumulative_bytes = 0
-    cumulative_slot_millis = 0
-    for name, results in results_dict.items():
-        if len(results) != 3:
-            raise IOError(
-                "Mismatch in performance logging output. "
-                "Expected one .bytesprocessed and one .slotmillis "
-                "file for each notebook."
-            )
-        query_count, total_bytes, total_slot_millis = results
-        cumulative_queries += query_count
-        cumulative_bytes += total_bytes
-        cumulative_slot_millis += total_slot_millis
-        print(
-            f"{name} - query count: {query_count},"
-            f" bytes processed sum: {total_bytes},"
-            f" slot millis sum: {total_slot_millis}"
-        )
-
-    print(
-        f"---total queries: {cumulative_queries}, "
-        f"total bytes: {cumulative_bytes}, "
-        f"total slot millis: {cumulative_slot_millis}---"
-    )
+    session.run("python", "scripts/publish_benchmark_result.py")
```

Comment:
Same here. Let's have the run and publish benchmark script take a file name.

Reply:
Done.